tests/kubernetes/test_pod_generator.py

```diff
@@ -684,7 +684,7 @@ def test_pod_name_is_valid(self, pod_id, expected_starts_with):
             len(name) <= 253 and all(ch.lower() == ch for ch in name) and re.match(regex, name)
         ), "pod_id is invalid - fails allowed regex check"
 
-        assert name.rsplit(".")[0] == expected_starts_with
+        assert name[:-33] == expected_starts_with
```

Comment:
```suggestion
        assert name.rsplit("-")[0] == expected_starts_with
```

Nit, I think the `rsplit` approach is easier to grok?

Reply:
My only worry is what if we change the "-" back to some other character... But then you could say the same worry about if we change the max pod length.

Reply:
Actually, this does create some issues. The test pod names are "pod-name-with-hyphen", so `name.rsplit("-")[0]` will just return `pod`. I switched it to use `"-".join(name.rsplit("-")[:-1])`. Unless we want to move back to the arbitrary `[:-33]` then this might be a better option.

Reply:
ref: https://github.com/apache/airflow/pull/19036/files#diff-60bb2e2d34655edf905c00b495e1feee30aac430040bfca7995c812841c84916R689

Reply:
> The test pod names are "pod-name-with-hyphen", so `name.rsplit("-")[0]` will just return `pod`.

Would `name.rsplit("-", 1)[0]` or `name.rpartition("-")[0]` work?

Reply:
`rsplit("-",1)[0]` works. Thanks for the tip. I am not as familiar with `rsplit`